[SMALLFIX] Using file path instead of file id to store a file in the UFS.

jsimsa opened the pull request:
This PR uses file path instead of file id to persist a file in the UFS. In addition, this PR fixes some minor  BlockDataManager formatting issues encountered while implementing the change.

yupeng9 commented on servers/src/main/java/tachyon/worker/block/BlockDataManager.java:
```diff
@@ -151,13 +151,22 @@ public void addCheckpoint(long sessionId, long fileId) throws TException, IOExce
-    String dstPath = PathUtils.concatPath(ufsDataFolder, fileId);
+    FileInfo fileInfo = mFileSystemMasterClient.getFileInfo(fileId);
+    TachyonURI uri = new TachyonURI(fileInfo.getPath());
+    String dstPath = PathUtils.concatPath(ufsDataFolder, fileInfo.getPath());
```
Could you add unit tests to cover this change?

jsimsa replied:
Is your concern code coverage or the missing unit test? There is currently no test class for the BlockDataManager (likely because testing the class requires non-trivial setup), but the code path I changed is covered by tests in the Tachyon Shell module. If your concern is the missing unit test, I am happy to create a proper unit test for BlockDataManager mocking all clients the manager uses.

yupeng9 replied:
Missing unit tests. Mocking sounds great!

jsimsa replied:
Done.

yupeng9 commented on servers/src/main/java/tachyon/worker/block/BlockDataManager.java:
```diff
@@ -151,13 +151,22 @@ public void addCheckpoint(long sessionId, long fileId) throws TException, IOExce
+    String dstPath = PathUtils.concatPath(ufsDataFolder, fileInfo.getPath());
+    String parentPath = PathUtils.concatPath(ufsDataFolder, uri.getParent().getPath());
     try {
+      if (!mUfs.exists(parentPath)) {
```
How about combine these two conditions?

jsimsa replied:
Done.

yupeng9 commented:
Thanks for the fix! I left some comments.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/4964/
Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/4965/
Test PASSed.

calvinjia commented:
LGTM

uronce-cc commented:
+1 LGTM

calvinjia commented:
@jsimsa Are you planning to add the unit test in this PR or another one?

jsimsa commented:
I am fine either way. (I have started working on it earlier today and expect to have something by EOD tomorrow).

jsimsa commented:
I have added some unit tests for the BlockDataManager. Note that I used Mockito for mocking various classes, which can only be applied to non-final classes. As a consequence, I have removed the "final" qualifier from all classes that the unit tests mock. PTAL

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/<h2>Failed Tests: <span class='status-failure'>29</span></h2><h3><a name='org.tachyonproject:tachyon-integration-tests' /><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport'>org.tachyonproject:tachyon-integration-tests</a>: <span class='status-failure'>29</span></h3><ul><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/IsolatedTachyonFileSystemIntegrationTest/lockBlockTest1/'><strong>tachyon.client.IsolatedTachyonFileSystemIntegrationTest.lockBlockTest1</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/IsolatedTachyonFileSystemIntegrationTest/lockBlockTest2/'><strong>tachyon.client.IsolatedTachyonFileSystemIntegrationTest.lockBlockTest2</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/IsolatedTachyonFileSystemIntegrationTest/lockBlockTest3/'><strong>tachyon.client.IsolatedTachyonFileSystemIntegrationTest.lockBlockTest3</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/IsolatedTachyonFileSystemIntegrationTest/unlockBlockTest1/'><strong>tachyon.client.IsolatedTachyonFileSystemIntegrationTest.unlockBlockTest1</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/IsolatedTachyonFileSystemIntegrationTest/unlockBlockTest2/'><strong>tachyon.client.IsolatedTachyonFileSystemIntegrationTest.unlockBlockTest2</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/IsolatedTachyonFileSystemIntegrationTest/unlockBlockTest3/'><strong>tachyon.client.IsolatedTachyonFileSystemIntegrationTest.unlockBlockTest3</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/skipTest_0_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.skipTest[0]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/skipTest_1_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.skipTest[1]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/skipTest_2_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.skipTest[2]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/skipTest_3_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.skipTest[3]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/readTest4_4_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.readTest4[4]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/readTest5_4_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.readTest5[4]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/readTest6_4_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.readTest6[4]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/skipTest_4_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.skipTest[4]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/readTest4_5_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.readTest4[5]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/readTest5_5_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.readTest5[5]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/readTest6_5_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.readTest6[5]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/skipTest_5_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.skipTest[5]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.master/ServiceSocketBindIntegrationTest/listenSameAddressTest/'><strong>tachyon.master.ServiceSocketBindIntegrationTest.listenSameAddressTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.master/ServiceSocketBindIntegrationTest/listenEmptyTest/'><strong>tachyon.master.ServiceSocketBindIntegrationTest.listenEmptyTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker/BlockServiceHandlerIntegrationTest/evictionTest/'><strong>tachyon.worker.BlockServiceHandlerIntegrationTest.evictionTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker/BlockServiceHandlerIntegrationTest/cancelBlockTest/'><strong>tachyon.worker.BlockServiceHandlerIntegrationTest.cancelBlockTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker/DataServerIntegrationTest/readMultiFiles_0_/'><strong>tachyon.worker.DataServerIntegrationTest.readMultiFiles[0]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker/DataServerIntegrationTest/readMultiFiles_1_/'><strong>tachyon.worker.DataServerIntegrationTest.readMultiFiles[1]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker/DataServerIntegrationTest/readMultiFiles_2_/'><strong>tachyon.worker.DataServerIntegrationTest.readMultiFiles[2]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker/DataServerIntegrationTest/readMultiFiles_3_/'><strong>tachyon.worker.DataServerIntegrationTest.readMultiFiles[3]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker/DataServerIntegrationTest/readMultiFiles_4_/'><strong>tachyon.worker.DataServerIntegrationTest.readMultiFiles[4]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker/DataServerIntegrationTest/readMultiFiles_5_/'><strong>tachyon.worker.DataServerIntegrationTest.readMultiFiles[5]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5023/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker.block.meta/TieredStoreIntegrationTest/unpinFileTest/'><strong>tachyon.worker.block.meta.TieredStoreIntegrationTest.unpinFileTest</strong></a></li></ul>
Test FAILed.

apc999 commented on servers/pom.xml:
```diff
@@ -103,6 +103,24 @@
       <scope>test</scope>
     </dependency>
     <dependency>
+      <groupId>org.hamcrest</groupId>
```
question: do we need these two dependencies on hamcrest?

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/<h2>Failed Tests: <span class='status-failure'>39</span></h2><h3><a name='org.tachyonproject:tachyon-integration-tests' /><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport'>org.tachyonproject:tachyon-integration-tests</a>: <span class='status-failure'>39</span></h3><ul><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/FileInStreamIntegrationTest/seekTest/'><strong>tachyon.client.FileInStreamIntegrationTest.seekTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/FileInStreamIntegrationTest/readEndOfFileTest/'><strong>tachyon.client.FileInStreamIntegrationTest.readEndOfFileTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/FileInStreamIntegrationTest/seekExceptionTest1/'><strong>tachyon.client.FileInStreamIntegrationTest.seekExceptionTest1</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/FileInStreamIntegrationTest/seekExceptionTest2/'><strong>tachyon.client.FileInStreamIntegrationTest.seekExceptionTest2</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/FileInStreamIntegrationTest/skipTest/'><strong>tachyon.client.FileInStreamIntegrationTest.skipTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/IsolatedTachyonFileSystemIntegrationTest/lockBlockTest1/'><strong>tachyon.client.IsolatedTachyonFileSystemIntegrationTest.lockBlockTest1</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/IsolatedTachyonFileSystemIntegrationTest/lockBlockTest2/'><strong>tachyon.client.IsolatedTachyonFileSystemIntegrationTest.lockBlockTest2</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/IsolatedTachyonFileSystemIntegrationTest/lockBlockTest3/'><strong>tachyon.client.IsolatedTachyonFileSystemIntegrationTest.lockBlockTest3</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/IsolatedTachyonFileSystemIntegrationTest/unlockBlockTest1/'><strong>tachyon.client.IsolatedTachyonFileSystemIntegrationTest.unlockBlockTest1</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/IsolatedTachyonFileSystemIntegrationTest/unlockBlockTest2/'><strong>tachyon.client.IsolatedTachyonFileSystemIntegrationTest.unlockBlockTest2</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/IsolatedTachyonFileSystemIntegrationTest/unlockBlockTest3/'><strong>tachyon.client.IsolatedTachyonFileSystemIntegrationTest.unlockBlockTest3</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/LocalBlockInStreamIntegrationTest/readTest1/'><strong>tachyon.client.LocalBlockInStreamIntegrationTest.readTest1</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/LocalBlockInStreamIntegrationTest/readTest2/'><strong>tachyon.client.LocalBlockInStreamIntegrationTest.readTest2</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/LocalBlockInStreamIntegrationTest/readTest3/'><strong>tachyon.client.LocalBlockInStreamIntegrationTest.readTest3</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/LocalBlockInStreamIntegrationTest/seekTest/'><strong>tachyon.client.LocalBlockInStreamIntegrationTest.seekTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/LocalBlockInStreamIntegrationTest/skipTest/'><strong>tachyon.client.LocalBlockInStreamIntegrationTest.skipTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/skipTest_0_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.skipTest[0]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/skipTest_1_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.skipTest[1]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/skipTest_2_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.skipTest[2]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/skipTest_3_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.skipTest[3]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/readTest4_4_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.readTest4[4]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/readTest5_4_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.readTest5[4]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/readTest6_4_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.readTest6[4]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/skipTest_4_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.skipTest[4]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/readTest4_5_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.readTest4[5]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/readTest5_5_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.readTest5[5]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/readTest6_5_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.readTest6[5]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.client/RemoteBlockInStreamIntegrationTest/skipTest_5_/'><strong>tachyon.client.RemoteBlockInStreamIntegrationTest.skipTest[5]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.master/ServiceSocketBindIntegrationTest/listenSameAddressTest/'><strong>tachyon.master.ServiceSocketBindIntegrationTest.listenSameAddressTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.master/ServiceSocketBindIntegrationTest/listenEmptyTest/'><strong>tachyon.master.ServiceSocketBindIntegrationTest.listenEmptyTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker/BlockServiceHandlerIntegrationTest/evictionTest/'><strong>tachyon.worker.BlockServiceHandlerIntegrationTest.evictionTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker/BlockServiceHandlerIntegrationTest/cancelBlockTest/'><strong>tachyon.worker.BlockServiceHandlerIntegrationTest.cancelBlockTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker/DataServerIntegrationTest/readMultiFiles_0_/'><strong>tachyon.worker.DataServerIntegrationTest.readMultiFiles[0]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker/DataServerIntegrationTest/readMultiFiles_1_/'><strong>tachyon.worker.DataServerIntegrationTest.readMultiFiles[1]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker/DataServerIntegrationTest/readMultiFiles_2_/'><strong>tachyon.worker.DataServerIntegrationTest.readMultiFiles[2]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker/DataServerIntegrationTest/readMultiFiles_3_/'><strong>tachyon.worker.DataServerIntegrationTest.readMultiFiles[3]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker/DataServerIntegrationTest/readMultiFiles_4_/'><strong>tachyon.worker.DataServerIntegrationTest.readMultiFiles[4]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker/DataServerIntegrationTest/readMultiFiles_5_/'><strong>tachyon.worker.DataServerIntegrationTest.readMultiFiles[5]</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5030/org.tachyonproject$tachyon-integration-tests/testReport/tachyon.worker.block.meta/TieredStoreIntegrationTest/unpinFileTest/'><strong>tachyon.worker.block.meta.TieredStoreIntegrationTest.unpinFileTest</strong></a></li></ul>
Test FAILed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5033/
Test PASSed.

yupeng9 commented on common/src/main/java/tachyon/test/Testable.java:
```diff
@@ -0,0 +1,44 @@
+ * }
+ * </code>
+ */
+public interface Testable<T> {
```
To be more precise, what do you think of PrivateMemberTestable?

jsimsa replied:
I would prefer to using simply Testable until we have multiple Testable interfaces to distinguish between. The JavaDoc explains what the interface does better than a verbose class name.

yupeng9 commented on servers/src/test/java/tachyon/worker/block/BlockDataManagerTest.java:
```diff
@@ -0,0 +1,199 @@
+  }
+
+  // TODO(jsimsa): Write unit tests for untested public methods.
+}
```
missing new line

jsimsa replied:
Done.

yupeng9 commented on servers/src/test/java/tachyon/worker/block/BlockDataManagerTest.java:
```diff
@@ -0,0 +1,199 @@
+
+  @Test
+  public void AbortBlockTest() throws Exception {
+    TestHarness harness = new TestHarness();
```
We can put it in before().

jsimsa replied:
Done.

yupeng9 commented on servers/src/test/java/tachyon/worker/block/BlockDataManagerTest.java:
```diff
@@ -0,0 +1,199 @@
+import tachyon.worker.block.meta.TempBlockMeta;
+
+public class BlockDataManagerTest {
+  class BlockDataManagerTester implements Tester<BlockDataManager> {
```
What do you think make lockDataManagerTest implement Tester, and call receiveAccess in before()?

jsimsa replied:
Great suggestion. Thanks!

yupeng9 commented:
Thanks for adding the interfaces for testable pattern! I left some comments.

jsimsa commented:
Addressed Yupeng's comments. PTAL

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5037/
Test PASSed.

yupeng9 commented on servers/src/test/java/tachyon/worker/block/BlockDataManagerTest.java:
```diff
@@ -0,0 +1,196 @@
+import tachyon.worker.block.meta.TempBlockMeta;
+
+public class BlockDataManagerTest implements Tester<BlockDataManager> {
+  TestHarness mHarness;
```
Question on convention: do we by default declare all the member variables as private?

jsimsa replied:
Done.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5039/
Test PASSed.

jsimsa commented:
Addressed your comment (and also avoided the need for removing the "final" qualifier). PTAL

yupeng9 commented:
That's awesome! Thanks for the improvement.

yupeng9 commented:
LGTM

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5041/<h2>Failed Tests: <span class='status-failure'>6</span></h2><h3><a name='org.tachyonproject:tachyon-servers' /><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5041/org.tachyonproject$tachyon-servers/testReport'>org.tachyonproject:tachyon-servers</a>: <span class='status-failure'>6</span></h3><ul><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5041/org.tachyonproject$tachyon-servers/testReport/tachyon.worker.block/BlockDataManagerTest/abortBlockTest/'><strong>tachyon.worker.block.BlockDataManagerTest.abortBlockTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5041/org.tachyonproject$tachyon-servers/testReport/tachyon.worker.block/BlockDataManagerTest/accessBlockTest/'><strong>tachyon.worker.block.BlockDataManagerTest.accessBlockTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5041/org.tachyonproject$tachyon-servers/testReport/tachyon.worker.block/BlockDataManagerTest/addCheckpointTest/'><strong>tachyon.worker.block.BlockDataManagerTest.addCheckpointTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5041/org.tachyonproject$tachyon-servers/testReport/tachyon.worker.block/BlockDataManagerTest/cleanupSessionsTest/'><strong>tachyon.worker.block.BlockDataManagerTest.cleanupSessionsTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5041/org.tachyonproject$tachyon-servers/testReport/tachyon.worker.block/BlockDataManagerTest/commitBlockTest/'><strong>tachyon.worker.block.BlockDataManagerTest.commitBlockTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5041/org.tachyonproject$tachyon-servers/testReport/tachyon.worker.block/BlockDataManagerTest/CreateBlockTest/'><strong>tachyon.worker.block.BlockDataManagerTest.CreateBlockTest</strong></a></li></ul>
Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5042/<h2>Failed Tests: <span class='status-failure'>6</span></h2><h3><a name='org.tachyonproject:tachyon-servers' /><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5042/org.tachyonproject$tachyon-servers/testReport'>org.tachyonproject:tachyon-servers</a>: <span class='status-failure'>6</span></h3><ul><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5042/org.tachyonproject$tachyon-servers/testReport/tachyon.worker.block/BlockDataManagerTest/abortBlockTest/'><strong>tachyon.worker.block.BlockDataManagerTest.abortBlockTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5042/org.tachyonproject$tachyon-servers/testReport/tachyon.worker.block/BlockDataManagerTest/accessBlockTest/'><strong>tachyon.worker.block.BlockDataManagerTest.accessBlockTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5042/org.tachyonproject$tachyon-servers/testReport/tachyon.worker.block/BlockDataManagerTest/addCheckpointTest/'><strong>tachyon.worker.block.BlockDataManagerTest.addCheckpointTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5042/org.tachyonproject$tachyon-servers/testReport/tachyon.worker.block/BlockDataManagerTest/cleanupSessionsTest/'><strong>tachyon.worker.block.BlockDataManagerTest.cleanupSessionsTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5042/org.tachyonproject$tachyon-servers/testReport/tachyon.worker.block/BlockDataManagerTest/commitBlockTest/'><strong>tachyon.worker.block.BlockDataManagerTest.commitBlockTest</strong></a></li><li><a href='https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5042/org.tachyonproject$tachyon-servers/testReport/tachyon.worker.block/BlockDataManagerTest/CreateBlockTest/'><strong>tachyon.worker.block.BlockDataManagerTest.CreateBlockTest</strong></a></li></ul>
Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5043/
Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5044/
Test FAILed.

jsimsa commented:
Jenkins, test this please.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5049/
Test PASSed.

jsimsa commented:
Resolved merge conflicts. PTAL

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5077/
Test PASSed.

yupeng9 commented:
LGTM

gpang commented on servers/src/main/java/tachyon/worker/block/BlockDataManager.java:
```diff
@@ -80,11 +84,12 @@
   public BlockDataManager(WorkerSource workerSource,
       WorkerBlockMasterClient workerBlockMasterClient,
-      WorkerFileSystemMasterClient workerFileSystemMasterClient) throws IOException {
+      WorkerFileSystemMasterClient workerFileSystemMasterClient, BlockStore blockStore)
```
Add the corresponding javadoc for the new blockStore parameter.

jsimsa replied:
Done.

gpang commented:
Just had a minor comment, but looks good! Thanks!

jsimsa commented:
Addressed Gene's comment. PTAL

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5086/
Test PASSed.

gpang commented:
LGTM
